server/src/train/trainvehiclelistitem.cpp

```diff
@@ -0,0 +1,111 @@
+/**
+ * server/src/train/trainvehiclelistitem.cpp
+ *
+ * This file is part of the traintastic source code.
+ *
+ * Copyright (C) 2023 Reinder Feenstra
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+#include "trainvehiclelistitem.hpp"
+#include "trainvehiclelist.hpp"
+#include "../world/getworld.hpp"
+#include "../core/attributes.hpp"
+#include "../core/objectproperty.tpp"
+#include "../vehicle/rail/railvehicle.hpp"
+
+
+TrainVehicleListItem::TrainVehicleListItem(TrainVehicleList &parent, uint32_t itemId) :
```

Comment:
I prefere to use the style:
```
Object::Object()
  : Something()
  , m_something
{
```

Because it results in less changes if things are added, you now don't need too add the trailing comma on the previous line :)

Reply:
It's not consistently used in other parts of code. We really should clang format everithing to keep consistency

Reply:
Older parts of the code base use the older style, clang-format would be best indeed, I did some experiments with it, but couldn't get it setup so it doesn't cause to much changes.